### PR TITLE
Improve error message

### PR DESCRIPTION
**Pull request by yiwen101**
Looks like a lot of files are edited, but that is because I merged with my previous PR as methods in it are needed.
Have improved the messages to user, and tested that for edit lesson, the following is a summary:

Screenshot 2023-11-01 at 12 23 25
Screenshot 2023-11-01 at 12 23 32
Screenshot 2023-11-01 at 12 23 36

### CODE REVIEW

**Comment by yiwen101** [comment]
Tested and updated add lesson and add person command. Following is summary:
Screenshot 2023-11-01 at 12 36 16
Screenshot 2023-11-01 at 12 43 11

**Comment by yiwen101** [comment]
Test and update edit person:
Screenshot 2023-11-01 at 12 53 07
Screenshot 2023-11-01 at 12 53 13

**Comment by yiwen101** [comment]
Fix #129

File: src/main/java/seedu/address/logic/commands/AddLessonCommand.java [hunk 1]
```diff
@@ -25,7 +25,7 @@ public CommandResult execute(Model model) throws CommandException {
 
         if (model.hasLessonClashWith(lesson)) {
             Lesson clashingLesson = model.getLessonClashWith(lesson);
-            throw new CommandException("Lesson already exists in the specified time slot: "
+            throw new CommandException("Exist lesson clashes with this lesson: "
```
Comment (richiehx) [inline src/main/java/seedu/address/logic/commands/AddLessonCommand.java:29]: I think Exist should be changed to Existing

File: src/main/java/seedu/address/logic/commands/AddPersonCommand.java [hunk 2]
```diff
@@ -75,7 +75,7 @@ public CommandResult execute(Model model) throws CommandException {
             // could execute without model by making model singleton
             if (model.hasLessonClashWith(lesson)) {
                 Lesson clashingLesson = model.getLessonClashWith(lesson);
-                throw new CommandException("Lesson already exists in the specified time slot: "
+                throw new CommandException("Exist lesson clashing with lesson to added: "
```
Comment (richiehx) [inline src/main/java/seedu/address/logic/commands/AddPersonCommand.java:79]: Same as previous comment, I think "Existing lesson clashes with lesson to be added" sounds better

File: src/main/java/seedu/address/logic/parser/EditLessonCommandParser.java [hunk 3]
```diff
@@ -14,14 +18,29 @@ public class EditLessonCommandParser implements Parser<EditLessonCommand> {
...
+        } else {
+            return new EditLessonCommand(lesson);
+        }
+    }
+
+    public static String getUsageInfo() {
+        return "\nUsage: edit <Index> (at least one of -[name|subject|day|start|end|remark] [value]). "
+                + "\nFor example, edit 1 -name lesson2 -subject Math -day Monday -start 14:30 -end 16:30"
+                + "\nIf you want to edit the currently shown lesson, you could omit the index. "
+                + "\nNote your edited 'name' must not already in the schedule and 'start' must be before 'end'.";
```
Comment (richiehx) [inline src/main/java/seedu/address/logic/parser/EditLessonCommandParser.java:44]: Suggestion to change to : Note your edited 'name' must not already be in the schedule and 'start' must be before 'end'.

File: src/main/java/seedu/address/logic/parser/EditPersonCommandParser.java [hunk 4]
```diff
@@ -15,10 +17,28 @@ public class EditPersonCommandParser implements Parser<EditPersonCommand> {
...
+            }
+        } else {
+            return new EditPersonCommand(person);
+        }
+    }
+    public static String getUsageInfo() {
+        return "\nUsage: edit <Index> (at least one of -[name|phone|email|address|subject|tag|remark] [value]). "
+                + "\nFor example, edit 1 -name John -phone 91234567"
+                + "\nIf you want to edit the currently shown person, you could omit the index. "
+                + "\nNote your edited 'name' must not already in the address book.";
```
Comment (richiehx) [inline src/main/java/seedu/address/logic/parser/EditPersonCommandParser.java:42]: Same as above.

**Review by richiehx — APPROVED** [review]
LGTM! Some minor grammatical comments

File: src/main/java/seedu/address/logic/commands/AddPersonCommand.java (hunk 2 above)
Comment (yiwen101) [reply src/main/java/seedu/address/logic/commands/AddPersonCommand.java:79]: Fix in new commit

File: src/main/java/seedu/address/logic/commands/AddLessonCommand.java (hunk 1 above)
Comment (yiwen101) [reply src/main/java/seedu/address/logic/commands/AddLessonCommand.java:29]: Fix in new commit

File: src/main/java/seedu/address/logic/parser/EditLessonCommandParser.java (hunk 3 above)
Comment (yiwen101) [reply src/main/java/seedu/address/logic/parser/EditLessonCommandParser.java:44]: Fix in new commit

File: src/main/java/seedu/address/logic/parser/EditPersonCommandParser.java (hunk 4 above)
Comment (yiwen101) [reply src/main/java/seedu/address/logic/parser/EditPersonCommandParser.java:42]: Fix in new commit

**Pull request merged**